### ISSUE TITLE
Help needed finding a test that changes behaviour when changing DefiningOpaqueTypes::No to Yes

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#127035]
cc @lcnr @compiler-errors 

I am trying to change

https://github.com/rust-lang/rust/blob/127fa2261b730a42e6d98b7927c3888ecd08f3e0/compiler/rustc_trait_selection/src/traits/select/confirmation.rs#L151-L201

and

https://github.com/rust-lang/rust/blob/127fa2261b730a42e6d98b7927c3888ecd08f3e0/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1759-L1828

to DefiningOpaqueTypes::Yes. But I have been unable to preserve an opaque type all the way to this point. I can provide some tests of previous attempts, but it's probably better to start from a clean slate ^^

Note that while I did add this change as a commit to https://github.com/rust-lang/rust/pull/127034/commits/2b1f4393399f2a28d3020c688a1063045f118086, the change can just be done on master without affecting any tests

**Comment by lcnr** [comment]
quite certain that the change in `confirm_projection_candidate` is only (modulo old solver weirdness) can only be triggered once we're using `DefineOpaqueTypes::Yes` here: https://github.com/rust-lang/rust/blob/0fdfb61795b40037955154638fe35f20b16c8e55/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1722

**Comment by oli-obk** [comment]
I did that in https://github.com/rust-lang/rust/pull/127034/commits/ced094520e3a492be3c635cecdada807478b5ec9, and tested everything on top of that

**Comment by rithishxcode** [comment]
@oli-obk Hi, is this issue solved? If not I would like to work on this

**Comment by oli-obk** [comment]
The changes have not been done yet, but I'm uncertain as to whether it is solveable or useful to be solved. Figuring out a test would be good regardless, but it's so far out of my cache I don't even know where to start rn

**Comment by lcnr** [comment]
yeah. I feel like it's fine to close this without finding such test. We'll stabilize TAIT using the new trait solver at which point a test for this would be nice, but not strictly necessary 🤔
